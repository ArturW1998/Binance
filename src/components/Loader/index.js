import React from 'react';

const Loader = () => (
  <div className="text-center w-100">
    <div className="lds-ellipsis">
      <div className="dot" />
      <div className="dot" />
      <div className="dot" />
      <div className="dot" />
    </div>
  </div>
);

export default Loader;
