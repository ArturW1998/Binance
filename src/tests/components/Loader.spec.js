import React from 'react';
import { shallow } from 'enzyme';
import Loader from '../../components/Loader';

describe('Loader component', () => {
  const loader = shallow(<Loader />);

  it('renders properly', () => {
    expect(loader.html()).toMatchSnapshot();
  });

  it('renders 4 dots', () => {
    expect(loader.find('.dot')).toHaveLength(4);
  });

  it('renders loader', () => {
    expect(loader.find('.lds-ellipsis')).toHaveLength(1);
  });
});
