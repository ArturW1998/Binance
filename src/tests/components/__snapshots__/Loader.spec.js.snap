// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader component renders properly 1`] = `"<div class=\\"text-center w-100\\"><div class=\\"lds-ellipsis\\"><div class=\\"dot\\"></div><div class=\\"dot\\"></div><div class=\\"dot\\"></div><div class=\\"dot\\"></div></div></div>"`;
